src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -276,16 +276,14 @@ function MoneyRequestAmountPage(props) {
     }, [props.iou.participants, props.iou.amount, props.iou.id]);
 
     useEffect(() => {
-        if (!props.route.params.currency) {
+        if (props.route.params.currency) {
+            setSelectedCurrencyCode(props.route.params.currency);
             return;
         }
-
-        setSelectedCurrencyCode(props.route.params.currency);
-    }, [props.route.params.currency]);
-
-    useEffect(() => {
-        setSelectedCurrencyCode(props.iou.currency);
-    }, [props.iou.currency]);
+        if (props.iou.currency) {
+            setSelectedCurrencyCode(props.iou.currency);
```

Comment:
A small flaw that I just noticed is that we're setting whatever currency that's passed in the query param.
eg: `/request/new?currency=ABC` would also work

We could have checked it against the list of valid currency codes stored in Onyx, but we can't do that because when deeplinking the local store isn't ready yet.

Reply:
it's in `tests/`. Let's not do that actually